components/chapters/LoginModal.tsx

```diff
@@ -0,0 +1,196 @@
+'use client'
+
+import Image from 'next/image'
+import { useState } from 'react'
+import Modal from 'react-modal'
+import crypto from 'crypto';
+import Secp256k1 from '@lionello/secp256k1-js';
+
+export const LoginModal = (props) => {
+  const [loaded, setLoaded] = useState(false);
+  const [userPrivateKey, setPrivateKey] = useState('');
+  const [user, setUser] = useState<any>({});
+  const [copied, setCopied] = useState(false);
+
+  function loadProgress() {
+    const privateKey = Secp256k1.uint256(Buffer.from(userPrivateKey, 'hex'), 16)
+    const publicKey = Secp256k1.generatePublicKeyFromPrivateKeyData(privateKey);
+
+    let user: any = window.localStorage.getItem('user')
+    if (user) {
+      user = JSON.parse(user)
+
+      if (user.publicKey.x === publicKey.x && user.publicKey.y === publicKey.y) {
+        window.localStorage.setItem('loggedIn', "true");
+        setUser(user);
+        setLoaded(true);
+      } else {
+        alert('Invalid private key')
+      }
+    }
+  }
+
+  function checkValidKey() {
+    if (userPrivateKey.length !== 64) return false;
+
+    const user = window.localStorage.getItem('user');
+    if (user) {
+      const userObj = JSON.parse(user);
+      const privateKey = Secp256k1.uint256(Buffer.from(userPrivateKey, 'hex'), 16)
+      const publicKey = Secp256k1.generatePublicKeyFromPrivateKeyData(privateKey);
+
+      if (userObj.publicKey.x === publicKey.x && userObj.publicKey.y === publicKey.y) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  function clearProgress() {
+    window.localStorage.removeItem('user');
+    window.localStorage.removeItem('loggedIn');
+
+    props.onClose();
+  }
+
+  function copy() {
+    navigator.clipboard.writeText(userPrivateKey)
+    setCopied(true)
+
+    setTimeout(() => {
+      setCopied(false)
+    }, 2000)
+  }
+
+  function signOut() {
+    setLoaded(false);
+    props.onLogout();
+    props.onClose();
+  }
+
+  return (
+    <Modal
+      isOpen={props.open}
+      className="top-1/2 left-1/2 outline-none p-5 transform -translate-x-1/2 -translate-y-1/2 absolute w-[32rem] rounded-lg shadow-lg bg-[#32455D] text-white"
```

Comment:
Instead of hard-coding colors, we need to find a way to assign dynamic colors via CSS variables, based on the color theme of the lesson ([example](https://github.com/GBKS/bitcoin-designathon/blob/main/assets/css/variables.scss#L10)). Seems like Tailwind has [support](https://tailwindcss.com/docs/adding-custom-styles#resolving-ambiguities), but we need to figure out how to assign these variables based on the current lesson.

For example, each content .mdx file could have a meta attribute called theme. That theme could be applied to the whole page as a class. That class could then adjust the variables accordingly.

Reply:
I introduced theme color variables in #45.

Reply:
Done